src/types.rs

```diff
@@ -386,3 +401,57 @@ pub struct VerifyResp {
     /// The block is pass or not.
     pub is_pass: bool,
 }
+
+///
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
+pub struct AggregatedChoke {
+    ///
+    pub chokes: Vec<SignedChoke>,
+    ///
+    pub signature: Signature,
+}
+
+#[allow(clippy::len_without_is_empty)]
+impl AggregatedChoke {
+    ///
+    pub fn len(&self) -> usize {
+        self.chokes.len()
+    }
+}
+
+///
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
+pub struct SignedChoke {
+    ///
+    pub signature: Signature,
+    ///
+    pub choke: Choke,
+    ///
+    pub address: Address,
+}
+
+/// A choke message.
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
+pub struct Choke {
```

Comment:
choke should not contain from, or you can not aggregate, because different nodes may have different from. you should wrap a new message contain choke which is metioned in design.